lib/Traits/ApiRequestTraits.ts

```diff
@@ -0,0 +1,59 @@
+import anyTrait from "./anyTrait";
+import ModelTraits from "./ModelTraits";
+import objectArrayTrait from "./objectArrayTrait";
+import primitiveTrait from "./primitiveTrait";
+import UrlTraits from "./UrlTraits";
+
+export class QueryParamTraits extends ModelTraits {
+  @primitiveTrait({
+    name: "Parameter name",
+    type: "string",
+    description: "The name of the query parameter."
+  })
+  name?: string;
+
+  @primitiveTrait({
+    name: "Parameter value",
+    type: "string",
+    description:
+      "The value of the query parameter. Parameter values starting with" +
+      " `DATE!`, eg. `DATE!HH:MM`, will be replaced wih the current date and" +
+      " time, formatted according to the format string following the `!`." +
+      " For more information on the format string format, see the " +
+      " [dateformat](https://github.com/felixge/node-dateformat) library."
+  })
+  value?: string;
+}
+
+export default class ApiRequestTraits extends UrlTraits {
```

Comment:
With this change, you've added `forceProxy` and `cacheDuration` - but they aren't implemented in `ApiTableCatalogItem`

Reply:
Lots of things that mixin UrlTraits don't implement either of those, eg. MapboxVectorTileCatalogItem, Cesium3DTilesCatalogItem. I thought the principle here was that just because a value is set, doesn't necessarily mean that something is done with that value.

Reply:
I don't think a Model should have a trait which isn't implemented. Maybe we should split up `UrlTraits` into `UrlTraits` and `UrlProxyTraits` or something

Reply:
This is also a later problem